sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample04_PublishingEvents.md

```diff
@@ -411,6 +411,8 @@ It is also important that you guard against exceptions in your handler code; it
 
 To ensure consistent performance and throughput, it is common for applications to make decisions around the pattern of publishing that they use - adjusting the frequency that batches are sent and how many operations take place concurrently.  Because the `EventHubBufferedProducerClient` manages batches and publishing in the background, your application cannot directly control these aspects.  
 
+Because the handlers are awaited, it is strongly advised that you *not* invoke `CloseAsync` or `DisposeAsync` from the handlers; doing so is likely to result in a deadlock scenario.  It is safe to attempt to resend events by adding them to the back of the buffer by calling `EnqueueEventAsync` or `EnqueueEventsAsync`
```

Comment:
Can we guard against this somehow?

Reply:
Not that I can think of without sacrificing the determinism of awaiting the handler before advancing to the next send.    If you've got ideas, I'm open to them.